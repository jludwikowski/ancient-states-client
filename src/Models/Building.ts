import BuildingPrototype from './BuildingPrototype';

export default interface Building extends BuildingPrototype{
    finalCost: number,
    finalBuildingTime: number,
    level: number
}
