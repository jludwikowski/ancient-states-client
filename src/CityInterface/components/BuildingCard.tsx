import React from 'react';
import Building from '../../Models/Building';

type Props = {
    building: Building,
    onBuild( id: number | undefined ): void;
} & React.HTMLAttributes<HTMLDivElement>;

export const BuildingCard: React.FunctionComponent<Props> = props => (
    <div className="card">
        <img className="card-img-top" src={props.building.imageUrl} />
        <div className="card-body">
            <h5 className="card-title">Name: {props.building.name}</h5>
            <div>Level: {props.building.level}</div>
            <div>Cost: {props.building.baseCost}</div>
            <div>Building time: {props.building.baseBuildingTime}</div>
        </div>
        <button className='btn btn-outline-secondary' onClick={() => props.onBuild(props.building.id)}>
            Build level {props.building.level+1}
        </button>
    </div>
)
