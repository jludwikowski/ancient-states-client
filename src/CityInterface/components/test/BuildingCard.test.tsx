import { shallow } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import Building from "../../../Models/Building";
import { BuildingCard } from '../BuildingCard';

describe('BuildingCard', () => {

    const building:Building = {
            baseBuildingTime: 120,
            baseCost: 100,
            description: 'You recruit standard units here',
            id: 1,
            imageUrl: 'images/icons/icons8-military-base-100.png',
            level: 10,
            name: 'barracks'
        };

    const onBuild = (id:number) => {
        console.log('On onDismiss');
    };

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<BuildingCard building={building} onBuild={onBuild} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Should match snapshot', () => {
        const component = shallow(<BuildingCard building={building} onBuild={onBuild} />);
        expect(component).toMatchSnapshot();
    });
});
