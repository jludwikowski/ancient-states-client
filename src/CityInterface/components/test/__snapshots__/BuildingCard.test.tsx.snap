// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuildingCard Should match snapshot 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top"
    src="images/icons/icons8-military-base-100.png"
  />
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Name: 
      barracks
    </h5>
    <div>
      Level: 
      10
    </div>
    <div>
      Cost: 
      100
    </div>
    <div>
      Building time: 
      120
    </div>
  </div>
  <button
    className="btn btn-outline-secondary"
    onClick={[Function]}
  >
    Build level 
    11
  </button>
</div>
`;
